### PR TITLE
[tests-only][full-ci] Add tests scenarios for updating role of public link to internal

### PR DESCRIPTION
**Pull request by roshanlc**
## Description
This PR adds test scenarios for updating role of a public link to internal.

## Related Issue

- Issue #4979 

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature [hunk 1]
```diff
@@ -610,3 +610,14 @@ Feature: update a public link share
...
+
+  Scenario: Updating the role of a public link to internal
+    Given user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/textfile.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | /textfile.txt |
+      | permissions | read          |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 0 |
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature:623]: Check if similar can be done here as well
```feature
Examples:
      | dav-path | ocs_api_version | ocs_status_code |
      | old      | 1               | 100             |
      | old      | 2               | 200             |
      | new      | 1               | 100             |
      | new      | 2               | 200             |
    @personalSpace
    Examples:
      | dav-path | ocs_api_version | ocs_status_code |
      | spaces   | 1               | 100             |
      | spaces   | 2               | 200             |
```

**Review by SagarGi — COMMENTED** [review]
others LGTM :+1:

**Review by saw-jan — COMMENTED** [review]
LGTM :+1: 

@SagarGi @roshanlc  

TODO after it gets merge 
- [ ] port to stable branch-2.0

**Review by SagarGi — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature [hunk 2]
```diff
@@ -610,3 +610,28 @@ Feature: update a public link share
...
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/textfile.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | /textfile.txt |
+      | permissions | read          |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 0 |
+    Then the OCS status code should be "<ocs_status_code>"
+    And the HTTP status code should be "200"
```
Comment (phil-davis) [inline tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature:625]: This is good, to test that the API call can be done to set the permissions to 0 (which means that anyone who accesses the pubic link will have to authenticate as a real user (i.e. it becomes a "link share" that is not really "public")

Are we also going to add API scenarios somewhere that confirm that the updated "link share" really does now work properly - a "public" API request to get the link share contents should fail, a request by a real user should pass.

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature (hunk 2 above)
Comment (ScharfViktor) [reply tests/acceptance/features/coreApiSharePublicLink3/updatePublicLinkShare.feature:625]: positive scenario we check in the e2e test: https://github.com/owncloud/web/pull/8260

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**